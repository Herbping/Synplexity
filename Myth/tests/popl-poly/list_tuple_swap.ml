type 'a list =
  | Nil
  | Cons of 'a * 'a list

let rec list_tuple_swap : ('a * 'b) list -> ('b * 'a) list |>
  /\([] -> [],
     [(a1, b1)] -> [(b1, a1)],
     [(a2, b2); (a1, b1)] -> [(b2, a2); (b1, a1)]) = ?
