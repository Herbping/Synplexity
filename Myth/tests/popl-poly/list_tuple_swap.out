(* Generated Thu Oct  8 20:27:09 2015 *)
fix f1 (l1:'a * 'b list) : 'b * 'a list ->
  match l1 with
  | Nil u1 -> Nil
  | Cons t1 -> Cons ((#2 (#1 t1), #1 (#1 t1)), f1 (#2 t1))
